### DISCUSSION TITLE
VPX Platform shoud expose rgb channels for "led" subtype

### DISCUSSION THREAD

**Discussion by mpcarr** [missionpinball/missionpinball#132] [General]
The VPX Platform should have the same parse light number to channels logic as the smart virtual platform so that light color information can be sent to vpx.

The led subtype should expose -r, -g, -b channels

**Comment by atummons** [comment]
I might be misinterpreting your request, but VPX has nothing to do with this tool. It's a completely separate item. There is a bridge to interface, but it has its own code and structure.

If that were to be desired, the request would be made to VPX to change their structure.

**Reply by mpcarr** [reply]
Thanks for the reply. 

See https://github.com/missionpinball/mpf/blob/098523700e4c3a04f1e88c9a058bef19a897c684/mpf/platforms/virtual_pinball/virtual_pinball.py#L376

Here all lights are mapped to one channel. My request is to make the led subtype have three channels for rgb. So that it has the same functionality as the "smart virtual" (mpf monitor) platform as here: https://github.com/missionpinball/mpf/blob/098523700e4c3a04f1e88c9a058bef19a897c684/mpf/platforms/virtual.py#L173

The vpx bridge will then receive brightness info for each channel allowing one to test rgb light changes.

**Comment by bernarma** [comment]
I have gone down the track of modifying the code to have separate rgb channels for RGB's. I had to make a few changes across both VPX (VBS) and MPF (python) but I have gotten it to work.

Although it works, from a performance standpoint, I would like to go back to using a singular value representing an RGB LED to minimise the chattiness that is a result of splitting the channels up. I.e. the interface currently sends one value per LED (which equates to 3 calls to VPX) - and the value ideally could be sent across in one call but split up according to some form of masking.

At the moment, I have spent enough time on it to get it to work correctly with the 3 values but at a later stage I will get stuck back into it. My "hack" was more a workaround to test the table so that I could visually see the correct LED color.

Since this topic is quite old, if you would like more details about this I can share it.
